pkg/builder/builder_test.go

```diff
@@ -86,21 +100,107 @@ func TestPrepareBuild(t *testing.T) {
 
-	if b.Version != mfst.Version {
-		t.Errorf("expected version %v, got %v", mfst.Version, b.Version)
+	if !reflect.DeepEqual(b.Credentials, mfst.Credentials) {
+		t.Errorf("expected credentials to be %+v but was %+v", mfst.Credentials, b.Credentials)
 	}
-	if b.SchemaVersion != mfst.SchemaVersion {
-		t.Errorf("expected schemaVersion %v, got %v", mfst.SchemaVersion, b.SchemaVersion)
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Custom, mfst.Custom) {
+		t.Errorf("expected custom to be %+v but was %+v", mfst.Custom, b.Custom)
+	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Definitions, mfst.Definitions) {
+		t.Errorf("expected definitions to be %+v but was %+v", mfst.Definitions, b.Definitions)
 	}
+	checksPerformed++
+
+	if b.Description != mfst.Description {
+		t.Errorf("expected description to be %+v but was %+v", mfst.Description, b.Description)
+	}
+	checksPerformed++
+
+	if len(b.InvocationImages) != 1 {
+		t.Fatalf("expected there to be 1 image, got %d. Full output: %v", len(b.Images), b)
+	}
+	checksPerformed++
+
 	expected := bundle.InvocationImage{}
 	expected.Image = "cnab:0.1.0"
 	expected.ImageType = "docker"
 	if !reflect.DeepEqual(b.InvocationImages[0], expected) {
 		t.Errorf("expected %v, got %v", expected, b.InvocationImages[0])
 	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Keywords, mfst.Keywords) {
+		t.Errorf("expected keywords to be %+v but was %+v", mfst.Keywords, b.Keywords)
+	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Maintainers, mfst.Maintainers) {
+		t.Errorf("expected maintainers to be %+v but was %+v", mfst.Maintainers, b.Maintainers)
+	}
+	checksPerformed++
+
+	if b.Name != mfst.Name {
+		t.Errorf("expected name to be %+v but was %+v", mfst.Name, b.Name)
+	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Outputs, mfst.Outputs) {
+		t.Errorf("expected outputs to be %+v but was %+v", mfst.Outputs, b.Outputs)
+	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Parameters, mfst.Parameters) {
+		t.Errorf("expected parameters to be %+v but was %+v", mfst.Parameters, b.Parameters)
+	}
+	checksPerformed++
+
+	if b.SchemaVersion != mfst.SchemaVersion {
+		t.Errorf("expected schemaVersion %v, got %v", mfst.SchemaVersion, b.SchemaVersion)
+	}
+	checksPerformed++
+
+	if b.Version != mfst.Version {
+		t.Errorf("expected version %v, got %v", mfst.Version, b.Version)
+	}
+	checksPerformed++
+
+	mfstFields := getFields(manifest.Manifest{})
+	if len(mfstFields) != checksPerformed {
+		t.Errorf("expected to check %v fields for equality, but checked only %v fields", len(mfstFields), checksPerformed)
+	}
+}
+
+func TestBundleAndManifestHasSameFields(t *testing.T) {
+	mfst := manifest.Manifest{}
+	mfstFields := getFields(mfst)
+
+	b := bundle.Bundle{}
+	bundleFields := getFields(b)
+
+	if !reflect.DeepEqual(bundleFields, mfstFields) {
+		t.Errorf("manifest and bundle have different fields.\nmanifest: %+v\nbundle: %+v\n", mfstFields, bundleFields)
+	}
+}
+
+func getFields(i interface{}) map[string]bool {
```

Comment:
It's conventional to use `struct{}` as a value type when representing sets as maps in Go.
```suggestion
func getFields(i interface{}) map[string]struct{} {
```